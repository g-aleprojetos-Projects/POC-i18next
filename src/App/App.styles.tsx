import styled from "styled-components";

export interface IApp {
  escolhido?: boolean
}

export const Container = styled.div`
display: flex;
flex-direction: column;
`

export const Header = styled.div`
display: flex;
justify-content: flex-end;
align-items: center;
width: 100%;
height: 100px;
background-color: #202024;
`

export const Nav = styled.nav`
display: flex;
padding-right: 30px;
`

export const Item = styled.a`
margin: 10px;
`
export const Texto = styled.p`
font-size: 15px;
color: #fff;
cursor: pointer;
`

export const Main = styled.div`
display: flex;
align-items: center;
justify-content: center;
`

export const Botao = styled.button<IApp>`
  width: 100px;
  height: 30px;
  margin: 10px;
  background-color: ${props => props.escolhido? "#aff": "#aaa" }
`
