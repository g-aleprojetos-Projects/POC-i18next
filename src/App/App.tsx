import React from 'react';
import * as S from './App.styles'
import { useTranslation } from 'react-i18next';
import { namespaces } from '../utils/i18n/i18n.constants';
import './App.css';
import '../utils/i18n';

export const App = () => {
  const {t, i18n} = useTranslation(namespaces.pages.header);

  return (
<S.Container>
   <S.Header>
     <S.Nav>
       <S.Item><S.Texto>{t("home")}</S.Texto></S.Item>
       <S.Item><S.Texto>{t("contents")}</S.Texto></S.Item>
       <S.Item><S.Texto>{t("about")}</S.Texto></S.Item>
     </S.Nav>
  </S.Header>
  <S.Main>    
        <S.Botao 
          escolhido={i18n.resolvedLanguage === "pt" }   
          type="submit" 
          onClick={() => i18n.changeLanguage("pt")}>
            PortuguÃªs
        </S.Botao>
        <S.Botao 
          escolhido={i18n.resolvedLanguage === "en" }   
          type="submit" 
          onClick={() => i18n.changeLanguage("en")}>
            English
        </S.Botao>      
  </S.Main>
</S.Container>
  );
}

