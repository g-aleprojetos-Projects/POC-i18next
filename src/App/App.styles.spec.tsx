import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import * as S from './App.styles'

describe('App.styles', () => {
   test('Container DEVE ser igual ao snapshot', () => {
      const container = renderer.create(<S.Container />).toJSON();
      expect(container).toMatchSnapshot();
    });
    test('Header DEVE ser igual ao snapshot', () => {
      const header = renderer.create(<S.Header />).toJSON();
      expect(header).toMatchSnapshot();
    });

    test('Nav DEVE ser igual ao snapshot', () => {
       const nav = renderer.create(<S.Nav />).toJSON();
       expect(nav).toMatchSnapshot();
    });

    test('Item DEVE ser igual ao snapshot', () => {
       const item = renderer.create(<S.Item />).toJSON();
       expect(item).toMatchSnapshot();
    });

    test('Texto DEVE ser igual ao snapshot', () => {
       const texto = renderer.create(<S.Texto />).toJSON();
       expect(texto).toMatchSnapshot();
    });

    test('Main DEVE ser igual ao snapshot', () => {
      const main = renderer.create(<S.Main />).toJSON();
      expect(main).toMatchSnapshot();
   });

   test('Botao DEVE ser igual ao snapshot', () => {
      const botao = renderer.create(<S.Botao />).toJSON();
      expect(botao).toMatchSnapshot();
   });

   test('Botao DEVE ser igual ao snapshot QUANDO "escolhido" for igual a true', () => {
      const botao = renderer.create(<S.Botao escolhido={true} />).toJSON();
      expect(botao).toMatchSnapshot();
   });
})