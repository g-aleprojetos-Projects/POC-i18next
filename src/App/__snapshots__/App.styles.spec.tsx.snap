// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.styles Botao DEVE ser igual ao snapshot 1`] = `
.c0 {
  width: 100px;
  height: 30px;
  margin: 10px;
  background-color: #aaa;
}

<button
  className="c0"
/>
`;

exports[`App.styles Botao DEVE ser igual ao snapshot QUANDO "escolhido" for igual a true 1`] = `
.c0 {
  width: 100px;
  height: 30px;
  margin: 10px;
  background-color: #aff;
}

<button
  className="c0"
/>
`;

exports[`App.styles Container DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
/>
`;

exports[`App.styles Header DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100px;
  background-color: #202024;
}

<div
  className="c0"
/>
`;

exports[`App.styles Item DEVE ser igual ao snapshot 1`] = `
.c0 {
  margin: 10px;
}

<a
  className="c0"
/>
`;

exports[`App.styles Main DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
/>
`;

exports[`App.styles Nav DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-right: 30px;
}

<nav
  className="c0"
/>
`;

exports[`App.styles Texto DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 15px;
  color: #fff;
  cursor: pointer;
}

<p
  className="c0"
/>
`;
